include/api.php

```diff
@@ -1689,33 +1678,24 @@ function api_statuses_home_timeline($type)
 
 	$start = $page * $count;
 
-	$sql_extra = '';
+	$condition = ["`uid` = ? AND `verb` = ? AND `item`.`id` > ?", api_user(), ACTIVITY_POST, $since_id];
+
 	if ($max_id > 0) {
-		$sql_extra .= ' AND `item`.`id` <= ' . intval($max_id);
+		$condition[0] .= " AND `item`.`id` <= ?";
+		$condition[] = $max_id;
 	}
 	if ($exclude_replies > 0) {
-		$sql_extra .= ' AND `item`.`parent` = `item`.`id`';
+		$condition[0] .= ' AND `item`.`parent` = `item`.`id`';
 	}
 	if ($conversation_id > 0) {
-		$sql_extra .= ' AND `item`.`parent` = ' . intval($conversation_id);
+		$condition[0] .= " AND `item`.`parent` = ?";
+		$condition[] = $conversation_id;
 	}
 
-	$r = q("SELECT `item`.* FROM `item`
```

Comment:
You removed the `$r` variable here but the foreach loop under still uses it.

Reply:
Where have you seen this loop? Could you point me to the line?

Reply:
Here: https://github.com/annando/friendica/blob/50dfc7d36ef08d1be8c50b0612ca0d8613c2aab5/include/api.php#L1702

Reply:
And fixed.